### ISSUE TITLE
Sentinel2 L1C data wrongly scaled

### ISSUE THREAD

**Issue by eomasters-repos** [CNES/sensorsio#1]
Hello, 

I think the Sentinel2-L1C data is wrongly scaled in the `read_as_numpy` function..

The `np_arr` contains the DN values and a scaled offset (default=-0.1) should not be applied.
I've noticed this when using sentinel2_superresolution and the values read where too low by 1000 (the scaling is set to 1).

Also, the no-data of the raw values is zero, but it is set to -10000 by default.

See page 444: https://sentinels.copernicus.eu/documents/247904/685211/S2-PDGS-TAS-DI-PSD-V14.9.pdf/3d3b6c9c-4334-dcc4-3aa7-f7c0deffbaf7?t=1643013091529

https://github.com/CNES/sensorsio/blob/b36e2b778ae290d82fbcd1e05122cd4b5ecb66a2/src/sensorsio/sentinel2_l1c.py#L272-L333

**Comment by jmichel-otb** [comment]
I agree on the `no_data` which should be set to 0 on line 321. I do not agree on removing the offset however. The purpose of `sensorsio` is to provide ready to use surface reflectance by default, not DN. According to the documentation you pointed out (p. 444), `RADIO_ADD_OFFSET` should be added if available in metadata (logic handled on line 118).

If you would like to be able to disable this behaviour we can consider an extra flag in `read_as_numpy()`, and a corresponding switch in `sentinel2_superresolution`.

**Comment by eomasters-repos** [comment]
I mean the offset should not be removed and also not the scaling.
But the scaling at Line#33 should not be:
`np_arr = np_arr + (self.radiometric_offset / scale)`
This is what is written in the PSD. But the PSD is wrong.
The formula results for a DN value of 5000 in 4999.9.
See here: https://sentiwiki.copernicus.eu/web/s2-products#S2Products-L1CS2-Products-L1Ctrue
Also discussed here: https://forum.step.esa.int/t/info-introduction-of-additional-radiometric-offset-in-pb04-00-products/35431/16?u=marco_eom

The line should be:
`np_arr = (np_arr + self.radiometric_offset) / scale`
which will give a value of 0.4

**Comment by jmichel-otb** [comment]
Ah, this is nasty. Thank you for spotting this. It will be fixed among other things in L1C driver (see https://framagit.org/jmichel-otb/sensorsio/-/merge_requests/14 ).

**Comment by jmichel-otb** [comment]
Fixes have been pushed to mr 14, branch is merged and github repo is synced.

Let me know if that works, so that I can close this issue.
